lib/snippets/kubernetes-deploy

```diff
@@ -50,6 +96,88 @@ class KubernetesDeploy
 
   private
 
+  def predeploy_priority_resources(resource_list)
+    PREDEPLOY_SEQUENCE.each do |resource_type|
+      matching_resources = resource_list.select { |r| r.type == resource_type }
+      next if matching_resources.empty?
+      deploy_resources(matching_resources)
+      wait_for_completion(matching_resources)
+    end
+  end
+
+  def discover_resources
+    resources = []
+    Dir.foreach(@template_path) do |filename|
+      next unless filename.end_with?(".yml.erb", ".yml")
+
+      split_templates(filename) do |tempfile|
+        resource_id = discover_resource_via_dry_run(tempfile)
+        type, name = resource_id.split("/", 2) # e.g. "pod/web-198612918-dzvfb"
+        resources << KubernetesResource.for_type(type, name, @namespace, tempfile)
+        KubernetesDeploy.logger.info "Discovered template for #{resource_id}"
+      end
+    end
+    resources
+  end
+
+  def discover_resource_via_dry_run(tempfile)
+    resource_id, err, st = run_kubectl("apply", "-f", tempfile.path, "--dry-run", "--output=name")
```

Comment:
The main script's `run_kubectl` always does automatically:
`KubernetesDeploy.logger.warn(err) unless st.success?`

Note that the resource instances' version doesn't though. My reasoning for that is that polling errors aren't blocking, and may of them will be 404s on GET for resources that aren't up yet... and that's not really an error as far as the script is concerned.